pkg/rancher-desktop/integrations/manageLinesInFile.ts

```diff
@@ -1,80 +1,135 @@
 import fs from 'fs';
-import os from 'os';
 
 import isEqual from 'lodash/isEqual.js';
 
 export const START_LINE = '### MANAGED BY RANCHER DESKTOP START (DO NOT EDIT)';
 export const END_LINE = '### MANAGED BY RANCHER DESKTOP END (DO NOT EDIT)';
 const DEFAULT_FILE_MODE = 0o644;
 
-// Inserts/removes fenced lines into/from a file. Idempotent.
-// @param path The path to the file to work on.
-// @param desiredManagedLines The lines to insert into the file.
-// @param desiredPresent Whether the lines should be present.
+/**
+ * Inserts/removes fenced lines into/from a file. Idempotent.
+ * @param path The path to the file to work on.
+ * @param desiredManagedLines The lines to insert into the file.
+ * @param desiredPresent Whether the lines should be present.
+ */
 export default async function manageLinesInFile(path: string, desiredManagedLines: string[], desiredPresent: boolean): Promise<void> {
-  // read file, creating it if it doesn't exist
-  let currentContent: string;
+  const copyFlags = fs.constants.COPYFILE_EXCL | fs.constants.COPYFILE_FICLONE;
+  const desired = getDesiredLines(desiredManagedLines, desiredPresent);
+  let fileStats: fs.Stats;
 
   try {
-    currentContent = await fs.promises.readFile(path, 'utf8');
-  } catch (error: any) {
-    if (error.code === 'ENOENT' && desiredPresent) {
-      const lines = buildFileLines([], desiredManagedLines, ['']);
-      const content = lines.join(os.EOL);
+    fileStats = await fs.promises.lstat(path);
+  } catch (ex: any) {
+    if (ex && 'code' in ex && ex.code === 'ENOENT') {
+      // File does not exist.
+      const content = computeTargetContents('', desired);
 
-      await fs.promises.writeFile(path, content, { mode: DEFAULT_FILE_MODE });
+      if (content) {
+        await fs.promises.writeFile(path, content, { mode: DEFAULT_FILE_MODE });
+      }
 
-      return;
-    } else if (error.code === 'ENOENT' && !desiredPresent) {
       return;
     } else {
-      throw error;
+      throw ex;
     }
   }
 
-  // split file into three parts
-  let before: string[];
-  let currentManagedLines: string[];
-  let after: string[];
+  if (fileStats.isFile()) {
+    if (await fileHasExtendedAttributes(path)) {
+      throw new Error(`Refusing to manage ${ path } which has extended attributes`);
+    }
 
-  try {
-    const currentLines = currentContent.split('\n');
+    const tempName = `${ path }.rd-temp`;
 
-    [before, currentManagedLines, after] = splitLinesByDelimiters(currentLines);
-  } catch (error) {
-    throw new Error(`could not split ${ path }: ${ error }`);
-  }
+    await fs.promises.copyFile(path, tempName, copyFlags);
```

Comment:
No, because we want to error out if the file already exists (indicating a race, or some leftover garbage from a previous crash).

The caller already needs to deal with this throwing.